deps/build.jl

```diff
@@ -26,11 +32,19 @@ if install_gap
     ## TODO: We currently use the GAP master branch.
     ##       Once all issues of using GAP with the julia
     ##       GC are resolved, we switch to a stable version.
-    run(`git clone https://github.com/gap-system/gap`)
+    run(`git clone --depth=1 https://github.com/gap-system/gap`)
     cd("gap")
     run(`./autogen.sh`)
     run(`./configure --with-gc=julia --with-julia=$(julia_binary)`)
     run(`make -j$(Sys.CPU_THREADS)`)
+    ## Hack to make GAPTypes available in the global package environment:
+    ## For a moment, we set the Julia load path to the actual depot, i.e.,
+    ## the env GAP.jl is installed to, to install GAPTypes.jl globally
+    julia_load_path_save = ENV["JULIA_LOAD_PATH"]
+    pop!( ENV, "JULIA_LOAD_PATH" )
+    julia_executable = joinpath(julia_binary, "julia")
+    run(`$(julia_executable) -e 'import Pkg; Pkg.add("GAPTypes")'`)
```

Comment:
Wow, that *is* a gross hack. I am also not quite sure why it is needed? Surely, GAP's `configure` also takes care of it? OK, perhaps the problem is that it inherits the `JULIA_LOAD_PATH`? But then at the very least, the comment above should explain so.

But I also wonder (very naively, so there is probably an easy answer why this won't work): couldn't we make the code in GAP more flexible, and ask it to first check if `GAP.GAPTypes` is available, then use that; if not, try to use `GAPTypes` directly?

Reply:
Yeah, I am painfully aware that this is gross, and should not be done this way. On the other hand, the Julia package manager sets this environment variable, too, which is also not very nice of it.

Actually, patching GAP this way would be possible, although not for the `4.10.2` release I guess. But then we can at least remove that hack afterwards.

Reply:
I will leave one more comment. The problem is that for the configure script, this somehow failed to give me the right responses when we asked for flags. I have no real idea why yet. This solution is merely the result of a lot of try-and-error.